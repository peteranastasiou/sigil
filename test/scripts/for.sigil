
echo "forward loop";
for i in 0:10 {
    print(i);
}

echo "reverse loop";
for i in 5 + 5 : 0 {
    print(i);
}

echo "expression in range";
fn ten() { 10 }
for i in 5 - 5 : ten() {
    print(i);
}

echo "implicit start value";
for i in 10 {
    print(i);
}

#for i in 1:=10 {
#    print(i);
#}

#for i in 10:0 {
#    print(i);
#}

#for i in 10:=0 {
#    print(i);
#}
