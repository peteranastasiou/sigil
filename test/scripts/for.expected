"forward loop"
0
1
2
3
4
5
6
7
8
9
"reverse loop"
10
9
8
7
6
5
4
3
2
1
"expression in range"
0
1
2
3
4
5
6
7
8
9
"implicit start value"
0
1
2
3
4
5
6
7
8
9
